### ISSUE TITLE
Stacey Generated 404 Cache Problem

### ISSUE THREAD

**Issue by bryandowning** [kolber/stacey#52]
This happened to me in Stacey 2.3.0

If there is no cache file for a stacey generated 404 page and you try to access a page that does not exist, you get a php error which gets cached and served even if you visit the 404 page directly.

```
Warning: file_get_contents(./templates/.) [function.file-get-contents]: failed to open stream: Permission denied in app\asset-types\page.inc.php on line 34
```
## Steps to recreate:

Obviously PHP error reporting needs to be on to see this in action.
1. Setup a stacey generated 404 page
   - create a 404 template file in `templates`
   - create a 404 directory in `content` with a `.txt` file pointing to the 404 template
   - modify `.htacces` `ErrorDocument` directive to say `ErrorDocument 404 /404`
2. Delete all files in `app/_cache` (except .gitignore)
3. Go to `http://base_url/some_page_that_does_not_exist` to see it error
4. Go to `http://base_url/404` to see the cached error
5. Delete all files in `app/_cache` (except .gitignore)
6. Go to `http://base_url/404` to see the 404 page
7. Go to `http://base_url/some_page_that_does_not_exist` to see the cached 404 page

**Comment by f0x2501** [comment]
Error in stacey 3, have error when I go to `http://base_url/some_page_that_does_not_exist`

But if I generate the page cache by acceding to `http://base_url/404` and go back to `http://base_url/some_page_that_does_not_exist` it's working.

**Comment by kolber** [comment]
Thanks for the heads up. I will have a look into this.

**Comment by kolber** [comment]
This should be fixed (in v3.0) with this commit: https://github.com/kolber/stacey/commit/70123824feae5d2b8d5e7cee54d1259bafb5badd

Can you confirm?

**Comment by bryandowning** [comment]
Works for me! Cheers!

**Comment by f0x2501** [comment]
Confirm the fix with 3.0 Thanks !
